bridge/pkg/terra/sender.go

```diff
@@ -0,0 +1,102 @@
+package terra
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/certusone/wormhole/bridge/pkg/terra/terra.go/client"
+	"github.com/certusone/wormhole/bridge/pkg/terra/terra.go/key"
+	"github.com/certusone/wormhole/bridge/pkg/terra/terra.go/msg"
+	"github.com/certusone/wormhole/bridge/pkg/terra/terra.go/tx"
+	"github.com/certusone/wormhole/bridge/pkg/vaa"
+)
+
+type JSONArraySlice []uint8
+
+func (u JSONArraySlice) MarshalJSON() ([]uint8, error) {
+	var result string
+	if u == nil {
+		result = "null"
+	} else {
+		result = strings.Join(strings.Fields(fmt.Sprintf("%d", u)), ",")
+	}
+	return []byte(result), nil
+}
+
+type SubmitVAAMsg struct {
+	Params SubmitVAAParams `json:"submit_v_a_a"`
+}
+
+type SubmitVAAParams struct {
+	VAA JSONArraySlice `json:"vaa"`
+}
+
+// SubmitVAA Prepares transaction with signed VAA and sends it to the Terra blockchain
+func SubmitVAA(urlLCD string, chainID string, contractAddress string, feePayer string, signed *vaa.VAA) (client.TxResponse, error) {
+
+	// Serialize VAA
+	vaa, err := signed.Marshal()
+	if err != nil {
+		return client.TxResponse{}, err
+	}
+
+	// Derive Raw Private Key
+	privKey, err := key.DerivePrivKey(feePayer, key.CreateHDPath(0, 0))
+	if err != nil {
+		return client.TxResponse{}, err
+	}
+
+	// Generate StdPrivKey
+	tmKey, err := key.StdPrivKeyGen(privKey)
+	if err != nil {
+		return client.TxResponse{}, err
+	}
+
+	// Generate Address from Public Key
+	addr := msg.AccAddress(tmKey.PubKey().Address())
+	if err != nil {
+		return client.TxResponse{}, err
+	}
+
+	// Create LCDClient
+	LCDClient := client.NewLCDClient(
```

Comment:
Is there a way to pass our runnable context here? What's the timeout? Can we get stuck here forever?